src/framework/components/rigid-body/system.js

```diff
@@ -3,33 +3,39 @@ import { ObjectPool } from '../../../core/object-pool.js';
 import { Debug } from '../../../core/debug.js';
 
 import { Vec3 } from '../../../core/math/vec3.js';
+import { Quat } from '../../../core/math/quat.js';
 
 import { Component } from '../component.js';
 import { ComponentSystem } from '../system.js';
 
-import { BODYFLAG_NORESPONSE_OBJECT } from './constants.js';
+import { BODYFLAG_NORESPONSE_OBJECT, BODYSTATE_ACTIVE_TAG } from './constants.js';
 import { RigidBodyComponent } from './component.js';
 import { RigidBodyComponentData } from './data.js';
 
-let ammoRayStart, ammoRayEnd;
+// Ammo.js variable for performance saving.
+let ammoRayStart, ammoRayEnd, ammoVec3, ammoQuat, ammoTransform, ammoTransform2;
+
+// RigidBody for shape tests. Permanent to save performance.
+let shapeTestBody;
```

Comment:
Any concerns with what happens if two instances of playcanvas are on the same page?  They'll share the same objects

Reply:
Many other files contain global variables to the module, I don't see the difference or problem here considering those variables are initialized once then every time a value is assigned to it, it is immediately used.

Do you suggest it could conflict with anything? And if so, why would it conflict here and not in any other module?

Reply:
We previously had issues with them hanging around an not being cleaned up properly.  More of a general point to be careful about module scope variables as they live forever and are shared between different instances.  I think in this case they are purely temporary variables so it should be fine but I thought it worth pointing out just in case